libraries/utilities.js

```diff
@@ -61,14 +61,24 @@ var ip = require('ip');       // get the device IP address library
 var dgram = require('dgram'); // UDP Broadcasting library
 var os = require('os');
 var path = require('path');
-
+const { spawn } = require('child_process');
 
 var hardwareInterfaces = {};
 
 var identityFolderName = '.identity'; // TODO: get this from server.js
 var homedir =  path.join(path.join(os.homedir(), 'Documents'), 'realityobjects');
 var hardwareIdentity = homedir + '/.identity';
 
+exports.restartServer = function () {
+  spawn("node", ["server"], {
+    detached: true,
+    stdio: 'inherit'  // [process.stdin, process.stdout, process.stderr]
+  }).unref()
+
+  process.kill(process.pid);
+  process.exit();
```

Comment:
What is the difference between these two lines?

Reply:
exit leaves the process, kill kills the app by its pid. kill seems to have a a wider reach when you spawn child processes.

You can remove process.exit(), but then again it never gets called and I left it in.